### DISCUSSION TITLE
Multiple named windows

### DISCUSSION THREAD

**Discussion by biggerlambda** [duckdb/duckdb#3622] [General]
I am writing a query where I want to create multiple named windows. So something like the following:
```
select avg(cost) over w30 , sum(cost) over w60 FROM table
window w30 AS ( order by timestamp asc rows between 30 preceding and 0 following )
window w60 AS (rows between 60 preceding and 0 following );
```
But I get an error:
```
Error: Parser Error: syntax error at or near "window"
LINE 3: window w60 AS ( rows between 60 preceding and 0 following )
```

Are multiple named windows supported in a single query ?

**Comment by Alex-Monahan** [comment]
Hello! Does separating them with a comma help?

**Reply by biggerlambda** [reply]
It doesn't.

**Reply by Alex-Monahan** [reply]
Oh! You only need the keyword `window` once. Then have a comma separated list of them.

**Reply by Alex-Monahan** [reply]
It's like a group by, for example. The clause name just needs to be there once, and expressions are comma separated after that!

**Reply by biggerlambda** [reply]
Thank you. That worked!
